### DISCUSSION TITLE
Schedule drop of PHP 7

### DISCUSSION THREAD

**Discussion by Alkarex** [FreshRSS/FreshRSS#5474] [Strategy]
Just like our older [discussion about the drop of PHP 5](https://github.com/FreshRSS/FreshRSS/discussions/3321), let's do the same for planning the drop of PHP 7.

As of FreshRSS [1.22.0](https://github.com/FreshRSS/FreshRSS/milestone/54), we support PHP 7.2+.

At the time of writing, PHP 7.4 is the most common version for WordPress https://wordpress.org/about/stats/#php_versions
Same according to w3techs statistics: https://w3techs.com/technologies/history_details/pl-php/all/q

But here are some notes already:

* Ubuntu 22.04 LTS supports PHP 8.1 https://packages.ubuntu.com/jammy/php
  * Ubuntu 20.04 LTS supports PHP 7.4 https://packages.ubuntu.com/focal/php until April 2025 (but will become 3rd oldest LTS by April 2024)
* Debian 12 supports PHP 8.2+ https://packages.debian.org/bookworm/php , but at the time of writing, has not yet landed in e.g. Raspberry Pi OS.
  * Debian 11 supports 7.4 https://packages.debian.org/bullseye/php until June 2024 (LTS until June 2026)
* Alpine 3.18 supports PHP 8.2
  * Alpine 3.16 supports PHP 8.1 https://pkgs.alpinelinux.org/packages?name=php81&branch=v3.16&arch=x86_64
* Red Hat RHEL-9 supports PHP 8.1
  * REHL-8 with PHP 8.0 ends its maintenance level 2 in June 2024.
* Fedora 37 supports PHP 8.1 https://packages.fedoraproject.org/pkgs/php/php/
* Synology DSM 7 supports PHP 8.2 https://www.synology.com/en-us/dsm/packages?search=php
    * DSM 6 is stuck to PHP 7.4 (might require a change of hardware for NAS models such as DS215j ?)
* FreeBSD 12 and 13 support PHP 8.1

There are plenty of interesting features to look forward to:
* https://www.php.net/releases/7_4_0.php
* https://www.php.net/releases/8.0/en.php
* https://www.php.net/releases/8.1/en.php

Based on this data, it seems to make sense to skip some PHP versions and jump from PHP 7.2 (now) to 7.4 (after Raspberry Pi OS 12, winter 2023) and then to 8.1 (after release of Ubuntu 24.04 LTS and after end of normal Debian 11 and REHL-8 support, and then awaiting maybe a bit some better usage statistics). That is to say skipping PHP 7.3 and then PHP 8.0 as well.
https://github.com/FreshRSS/FreshRSS/milestones

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/5720

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/pull/6711

**Comment by Alkarex** [comment]
Related information from Packagist for SimplePie (which does not include the many servers running WordPress): PHP 8.1+ accounts for over 71% of downloads in September 2024:
https://packagist.org/packages/simplepie/simplepie/php-stats
![image](https://github.com/user-attachments/assets/d19501ec-1cac-4b92-bc0c-2a07aeb92dbf)

**Comment by Alkarex** [comment]
Note for (much) later: there does not seem to be any new feature in PHP 8.2 or 8.3 to justify an increase of requirements from the 8.1+ that we now have, but there are some very relevant new features in PHP 8.4 (native HTML5 and CSS selectors), so we will probably jump from 8.1+ to 8.4+ when the time comes (not before ~2028)
https://github.com/FreshRSS/FreshRSS/issues/6831

**Reply by math-GH** [reply]
> ..., so we will probably jump from 8.1+ to 8.4+ when the time comes (not before ~2028)

not before ~2028?
Milestone 1.26.0 announce 8.4+: https://github.com/FreshRSS/FreshRSS/milestones

**Reply by Alkarex** [reply]
> Milestone 1.26.0 announce 8.4+

Yes, to start taking advantage of some 8.4 features, but that does not mean dropping previous versions just yet :-)

**Reply by Alkarex** [reply]
A few notes for later:

* PHP upstream supports PHP 8.5 until December 2029 https://www.php.net/supported-versions.php
	* PHP 8.4 until December 2028
	* PHP 8.3 until December 2027
	* PHP 8.2 until December 2026
* Ubuntu 26.04 LTS supports PHP 8.5 until May 2031 https://packages.ubuntu.com/resolute/php
	* Ubuntu 24.04 LTS supports PHP 8.3 until June 2029 https://packages.ubuntu.com/noble/php
	* Ubuntu 22.04 LTS supports PHP 8.1 until June 2027 https://packages.ubuntu.com/jammy/php
* Debian Trixie supports PHP 8.4 until August 2028 (LTS June 2030) https://packages.debian.org/trixie/php
	* Debian Bookworm supports PHP 8.2 until June 2026 (LTS June 2028) https://packages.debian.org/bookworm/php
* Alpine 3.23 supports PHP 8.5 (and 8.4, 8.3) until November 2027 https://pkgs.alpinelinux.org/packages?name=php%3F%3F&branch=v3.23&arch=x86_64
	* Alpine 3.22 supports PHP 8.4 (and 8.3, 8.2) until May 2027 https://pkgs.alpinelinux.org/packages?name=php%3F%3F&branch=v3.22&arch=x86_64
	* Alpine 3.21 supports PHP 8.4 (and 8.3, 8.2) until November 2026 https://pkgs.alpinelinux.org/packages?name=php%3F%3F&branch=v3.21&arch=x86_64
	* ~~Alpine 3.20 supports PHP 8.3 (and 8.2) until April 2026 https://pkgs.alpinelinux.org/packages?name=php%3F%3F&branch=v3.20&arch=x86_64~~
* Fedora 44 supports PHP 8.5 until June 2027 https://packages.fedoraproject.org/pkgs/php/php/
	* Fedora 43 supports PHP 8.4 until December 2026
	* Fedora 42 supports PHP 8.4 until May 2026
* Red Hat RHEL-10 supports PHP 8.3 until 2035
	* Red Hat RHEL-9 supports PHP 8.3 until 2032
	* Red Hat RHEL-8 supports PHP 8.2 until 2029
* Synology DSM 7 supports PHP 8.4 https://www.synology.com/en-us/dsm/packages?search=php
